source/features/action-pr-link.tsx

```diff
@@ -2,25 +2,49 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import observe from '../helpers/selector-observer.js';
+import select from "select-dom";
+import GitHubURL from "../github-helpers/github-url.js";
 
-async function add(prLink: HTMLAnchorElement): Promise<void> {
+function setSearchParam(anchorElement: HTMLAnchorElement, name: string, value: string) {
+	const params = new URLSearchParams(anchorElement.search);
+	params.set(name, value)
+	anchorElement.search = String(params)
+}
+
+async function addForRepositoryActions(prLink: HTMLAnchorElement): Promise<void> {
 	const prNumber = prLink.textContent!.slice(1);
 
 	const runLink = prLink.closest('.Box-row')!.querySelector('a.Link--primary')!;
-	const parameters = new URLSearchParams(runLink.search);
-	parameters.set('pr', prNumber);
-	runLink.search = String(parameters);
+	setSearchParam(runLink, 'pr', prNumber)
+}
+
+async function addForPR(dropdown: HTMLDivElement): Promise<void> {
+	const {branch: prNumber} = new GitHubURL(location.href);
+	const actions = select.all('div.merge-status-item:has(a[href="/apps/github-actions"]) a.status-actions', dropdown)
```

Comment:
I suggest selecting the links in the `observe`, not by selecting the dropdown first